src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -813,20 +823,12 @@ std::string GcsServer::GetDebugState() const {
   return stream.str();
 }
 
-std::shared_ptr<RedisClient> GcsServer::GetOrConnectRedis() {
-  if (redis_client_ == nullptr) {
-    redis_client_ = std::make_shared<RedisClient>(GetRedisClientOptions());
-    auto status = redis_client_->Connect(io_context_provider_.GetDefaultIOContext());
-    RAY_CHECK(status.ok()) << "Failed to init redis gcs client as " << status;
-
-    // Init redis failure detector.
-    gcs_redis_failure_detector_ = std::make_unique<GcsRedisFailureDetector>(
-        io_context_provider_.GetDefaultIOContext(), redis_client_, []() {
-          RAY_LOG(FATAL) << "Redis connection failed. Shutdown GCS.";
-        });
-    gcs_redis_failure_detector_->Start();
-  }
-  return redis_client_;
+std::shared_ptr<RedisClient> GcsServer::CreateRedisClient(
+    instrumented_io_context &io_service) {
+  auto redis_client = std::make_shared<RedisClient>(GetRedisClientOptions());
+  auto status = redis_client->Connect(io_service);
+  RAY_CHECK(status.ok()) << "Failed to init redis gcs client as " << status;
```

Comment:
I see `RAY_CHECK(status.ok())` quite a lot in our codebase, do you think it's better to have a macro `RAY_CHECK_STATUS`?

Reply:
we have RAY_CHECK_OK, later we can do a repo-wide refactor for it

Reply:
Gotcha, thanks for the info!